File: Backend/HRMS/HRMS.Application/DTOs/Core/CityDto.cs
namespace HRMS.Application.DTOs.Core;

/// <summary>
/// DTO لعرض بيانات المدينة الكاملة
/// </summary>
public class CityDto
{
    public int CityId { get; set; }
    public int CountryId { get; set; }
    public string CityNameAr { get; set; } = string.Empty;
    public string CityNameEn { get; set; } = string.Empty;
    public string CountryNameAr { get; set; } = string.Empty;
    public string CountryNameEn { get; set; } = string.Empty;
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: Frontend/HRMS_Frontend/src/app/features/setup/models/city.dto.ts
export interface CityDto {
  cityId: number;
  countryId: number;
  cityNameAr: string;
  cityNameEn: string;
  countryNameAr: string;
  countryNameEn: string;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CityListDto {
  cityId: number;
  cityNameAr: string;
  cityNameEn: string;
  countryNameAr: string;
  isActive: boolean;
}

export interface CreateCityCommand {
  cityNameAr: string;
  cityNameEn: string;
  countryId: number;
}

export interface UpdateCityCommand {
  cityId: number;
  countryId: number;
  cityNameAr: string;
  cityNameEn: string;
  isActive: boolean;
}
